### PR TITLE
fixes #25224 - show arch from product as backup

### PR DESCRIPTION
**Pull request by stbenjam**
If there's no basearch substitution, we can check and see if Candlepin knows the arch for the product. Under the new CDN structure, repository sets will only have 1 arch and no substitution variable.  So, this PR makes a differentiation between the display architecture in the UI, and an architecture substitution.  I'm not sure if this is the best approach, maybe someone familiar with the repos page has a better idea of how to fix this.

To test -

1. Use a manifest that has beta products in it, and hit the stage CDN
2. Notice repo sets that have arch in the name show "Undefined" when trying to enable them
3. Apply this patch
4. Notice that it correctly shows the architecture in the  list

### CODE REVIEW

File: app/models/katello/glue/candlepin/product.rb [hunk 1]
```diff
@@ -138,7 +138,7 @@ def default_arch
 
       def convert_from_cp_fields(cp_json)
         ar_safe_json = cp_json.key?(:attributes) ? cp_json.merge(:attrs => cp_json.delete(:attributes)) : cp_json
-        ar_safe_json[:attrs] = remove_hibernate_fields(cp_json[:attrs]) if ar_safe_json.key?(:attrs)
+        ar_safe_json[:attrs] = remove_hibernate_fields(ar_safe_json[:attrs]) if ar_safe_json.key?(:attrs)
```
Comment (stbenjam) [inline app/models/katello/glue/candlepin/product.rb]: This is fixing a bug - convert_from_cp_fields is otherwise broken

**Comment by stbenjam** [comment]
[test prprocessor]

**Comment by johnpmitsch** [comment]
@stbenjam getting an error with this PR when trying to expand `Red Hat Enterprise Linux 5 Server - Extended Life Cycle Support (RPMs)` and not seeing the same on master:
```
RepositorySetRepositories.js:33 Uncaught TypeError: Cannot read property 'split' of undefined
    at https://centos7-devel.jomitsch.example.com:3808/webpack/katello.js:27715:50
    at Array.sort (native)
    at RepositorySetRepositories._this.sortedRepos (https://centos7-devel.jomitsch.example.com:3808/webpack/katello.js:27702:12)
    at RepositorySetRepositories.render (https://centos7-devel.jomitsch.example.com:3808/webpack/katello.js:27766:33)
    at finishClassComponent (https://centos7-devel.jomitsch.example.com:3808/webpack/vendor.js:144994:31)
    at updateClassComponent (https://centos7-devel.jomitsch.example.com:3808/webpack/vendor.js:144957:10)
    at beginWork (https://centos7-devel.jomitsch.example.com:3808/webpack/vendor.js:145546:16)
    at performUnitOfWork (https://centos7-devel.jomitsch.example.com:3808/webpack/vendor.js:147872:12)
    at workLoop (https://centos7-devel.jomitsch.example.com:3808/webpack/vendor.js:147910:24)
    at HTMLUnknownElement.callCallback (https://centos7-devel.jomitsch.example.com:3808/webpack/vendor.js:131601:14)
```

**Comment by stbenjam** [comment]
I guess the problem with my approach is there's so many assumptions on this page that if an architecture exists, then it MUST be used to substitute for basearch in the URL.  I tried to split that into a display vs. substitution arch but I seemed to be breaking other repos.  

My backup approach was to just change 'Unspecified' to 'Default' - if a repository set has no arch then just show Default as the one to enable, as the architecture is already visible in the repository set's name. I don't think that's a great solution. Maybe we don't expand the menu at all and just show a + if a repository set doesn't have arch variants?

**Comment by beav** [comment]
Personally, I'm OK with using "Default" for now and creating a new issue, assuming only one "Default" can exist at a time. It is less unsightly than "Unspecified" and it sounds like fetching the correct arch may need some design for it to work correctly in all cases.

**Comment by stbenjam** [comment]
I think @johnpmitsch was going to try to find some time to take a look at a better solution, otherwise, we can get this in - whatever works best for you all.

**Comment by johnpmitsch** [comment]
@stbenjam I opened up a PR here https://github.com/Katello/katello/pull/7882, feel free to close this one.

**Pull request closed without merging**